flocker/restapi/docs/publicapi.py

```diff
@@ -109,33 +109,34 @@ def _parseSchema(schema, schema_store):
         document the schema.
     """
     result = {}
+    schema = resolveSchema(schema, schema_store)
 
-    resolver = LocalRefResolver(
-        base_uri=b'',
-        referrer=schema, store=schema_store)
-
-    if schema.get(u'$ref') is None:
-        raise Exception('Non-$ref top-level definitions not supported.')
-
-    def fill_in_attribute(attr, propSchema):
-        attr['title'] = propSchema['title']
-        attr['description'] = prepare_docstring(
-            propSchema['description'])
-        attr['required'] = property in schema.get('required', [])
-        attr['type'] = propSchema['type']
-
-    with resolver.resolving(schema[u'$ref']) as schema:
-        if schema[u'type'] != u'object':
-            raise Exception('Non-object top-level definitions not supported.')
-
+    def fill_in_result(object_schema):
         result['properties'] = {}
-        for property, propSchema in schema[u'properties'].iteritems():
+        for property, propSchema in object_schema[u'properties'].iteritems():
             attr = result['properties'][property] = {}
-            if "$ref" in propSchema:
-                with resolver.resolving(propSchema['$ref']) as propSchema:
-                    fill_in_attribute(attr, propSchema)
-            else:
-                fill_in_attribute(attr, propSchema)
+            attr['title'] = propSchema['title']
+            attr['description'] = prepare_docstring(
+                propSchema['description'])
+            attr['required'] = property in object_schema.get("required", [])
+            attr['type'] = propSchema['type']
+
+    if schema[u"type"] == u"object":
+        result["type"] = "object"
+        fill_in_result(schema)
+    elif schema[u"type"] == u"array":
+        result["type"] = "array"
+        # Assume children of array are objects, and there's only one
```

Comment:
Comment here that this is for ease of implementation now and not because arrays cannot be arbitrarily complex. This means that someone adding a more complex JSON schema later can add a feature to the code here.